src/modules/convex/type/model.hpp

```diff
@@ -126,45 +129,96 @@ struct MLPModel {
         for (k = 0; k < N; k ++) {
             size += static_cast<uint32_t>((n[k] + 1) * (n[k+1]));
         }
+        //TODO conditionally assign size based on momentum
```

Comment:
yes it is, right now we create the velocity vector even if momentum is set to 0. Ideally we should only create the velocity vector if momentum>0. This would mean that the model size would be just `size` for 0 momentum and `size*2` otherwise. 
We can address this TODO in a future commit